front/src/modules/timesStops/hooks/useComputedMargins.tsx

```diff
@@ -0,0 +1,73 @@
+import type { SimulationResponseSuccess } from 'applications/operationalStudies/types';
+import type { TrainScheduleResult } from 'common/api/generatedEditoastApi';
+import type { OperationalPointWithTimeAndSpeed } from 'modules/trainschedule/components/DriverTrainScheduleV2/types';
+import { interpolateValue } from 'modules/trainschedule/components/DriverTrainScheduleV2/utils';
+
+import { marginRegExValidation } from '../consts';
+import { formatDigitsAndUnit } from '../utils';
+
+let marginUnit = '%';
+
+const useComputedMargins = (
+  trainSimulation: SimulationResponseSuccess,
+  opPoint: OperationalPointWithTimeAndSpeed,
+  nextOpPoint: OperationalPointWithTimeAndSpeed,
+  selectedTrainSchedule: TrainScheduleResult,
+  pathId: string
+) => {
+  const theoreticalMarginIndex = selectedTrainSchedule?.margins?.boundaries?.findIndex(
+    (id) => id === pathId
+  );
+  if (!theoreticalMarginIndex) {
```

Comment:
Early return if:
- !theoreticalMarginIndex
- theoreticalMarginIndex is -1 or is the last one
- selectedTrainSchedule has no margins (maybe this test can be done before the call to the function)